portlet-parent/wicketstuff-portlet/src/main/java/org/apache/wicket/portlet/PortletUrlRenderer.java

```diff
@@ -1,33 +1,57 @@
 package org.apache.wicket.portlet;
 
+import javax.portlet.PortletRequest;
+
 import org.apache.wicket.request.Request;
 import org.apache.wicket.request.UrlRenderer;
 import org.apache.wicket.util.lang.Args;
 import org.apache.wicket.util.string.PrependingStringBuffer;
 
 /**
- * {@link UrlRenderer} for Portlet Applications.
+ * <h1>{@link UrlRenderer} for Portlet Applications.</h1>
+ * <p>
+ * It corrects the base Url. All the generated URLs will be relative to this Url.
+ * </p>
+ * <p>
+ * The base Url depends on the lifecycle phase of the current portlet request.
+ * </p>
+ * <ul>
+ * <li>During the RENDER or the ACTION phase the base url must be set with the request url which is
+ * relative to the wicket filter path {@link Request#getUrl()}.  

```

Comment:
Hello Martin,

I am sorry for the confusion due to my comments.

2016-05-30 18:28 GMT+03:00 Martin Grigorov <notifications@github.com>:

> In
> portlet-parent/wicketstuff-portlet/src/main/java/org/apache/wicket/portlet/PortletUrlRenderer.java
> <https://github.com/wicketstuff/core/pull/503#discussion_r65085638>:
>
> >  import org.apache.wicket.request.Request;
> >  import org.apache.wicket.request.UrlRenderer;
> >  import org.apache.wicket.util.lang.Args;
> >  import org.apache.wicket.util.string.PrependingStringBuffer;
> >
> >  /**
> > - * {@link UrlRenderer} for Portlet Applications.
> > + * <h1>{@link UrlRenderer} for Portlet Applications.</h1>
> > + * <p>
> > + * It corrects the base Url. All the generated URLs will be relative to this Url.
> > + * </p>
> > + * <p>
> > + * The base Url depends on the lifecycle phase of the current portlet request.
> > + * </p>
> > + * <ul>
> > + * <li>During the RENDER or the ACTION phase the base url must be set with the request url which is
> > + * relative to the wicket filter path {@link Request#getUrl()}.
>
> OK. This answers my question - there is no difference between RENDER and
> ACTION.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/wicketstuff/core/pull/503/files/4004ec2d893dfdcee23c40ee67f778a1510f0775#r65085638>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AP5KvbPvd9Gp3a0r7pWgdCrY-Hmusmffks5qGwIdgaJpZM4IpMQK>
> .
>